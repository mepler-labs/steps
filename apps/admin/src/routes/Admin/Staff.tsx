import { addAlert } from 'actions/alerts';
import { hideModal, showModal } from 'actions/modals';
import {
  deleteCoach,
  getCoaches,
  resendInvite,
  updatePermissions,
} from 'actions/staff';
import Button from 'atoms/Buttons/Button';
import Main from 'atoms/Main';
import { AlertLevel } from 'components/Alert/types';
import PageHeader from 'components/Headers/PageHeader';
import StaffList from 'components/StaffList/StaffList';
import Modal from 'containers/Modal';
import React from 'react';
import { connect } from 'react-redux';
import { User, USER_TYPE } from 'reducers/auth';
import { bindActionCreators } from 'redux';
import AdminNewStaff, { NEW_STAFF } from './NewStaff';
import { ModalSize } from 'components/Modal';

interface Props {
  coaches: User[];
  actions: {
    getCoaches;
    addAlert;
    deleteCoach;
    resendInvite;
    updatePermissions;
    hideModal;
    showModal;
  };
}

export class Staff extends React.Component<Props> {
  componentWillMount() {
    this.props.actions.getCoaches().catch(err => {
      this.props.actions.addAlert({
        level: AlertLevel.Error,
        message: err.message,
        id: 'get-coaches-error',
      });
    });
  }

  onDelete = (coach: User) => {
    this.props.actions.deleteCoach(coach.id).catch(err => {
      this.props.actions.addAlert({
        level: AlertLevel.Error,
        message: err.message,
        id: 'delete-coach-error',
      });
    });
    // TODO: This only works for coaches with an ID (not invited coaches). We can change this to use an email, or give invited coaches an ID.
  };

  onResend = (coach: User) => {
    this.props.actions.resendInvite(coach.email).catch(err => {
      this.props.actions.addAlert({
        level: AlertLevel.Error,
        message: err.message,
        id: 'resend-invite-error',
      });
    });
  };

  onUpdateRole = (role: USER_TYPE, coach: User) => {
    this.props.actions.updatePermissions(coach.id, role).catch(err => {
      this.props.actions.addAlert({
        level: AlertLevel.Error,
        message: err.message,
        id: 'update-permissions-error',
      });
    });
  };

  render() {
    const { actions } = this.props;
    return (
      <Main>
        <PageHeader label="Staff">
          <Button onClick={() => actions.showModal(NEW_STAFF)}>
            Invite Staff
          </Button>
        </PageHeader>
        <StaffList
          onDelete={this.onDelete}
          onResend={this.onResend}
          onUpdateRole={this.onUpdateRole}
          staff={this.props.coaches}
        />
        <Modal
          id={NEW_STAFF}
          onClose={() => actions.hideModal(NEW_STAFF)}
          size={ModalSize.Medium}
        >
          <AdminNewStaff />
        </Modal>
      </Main>
    );
  }
}

const mapStateToProps = ({ staff }) => ({
  coaches: staff.coaches,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      getCoaches,
      addAlert,
      deleteCoach,
      resendInvite,
      updatePermissions,
      showModal,
      hideModal,
    },
    dispatch,
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Staff);
