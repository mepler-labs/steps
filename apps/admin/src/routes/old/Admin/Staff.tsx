import Button from 'atoms/Buttons/Button';
import Label from 'atoms/Label';
import Main from 'atoms/Main';
import PageHeader from 'components/Headers/PageHeader';
import StaffList from 'components/StaffList/StaffList';
import { StaffMember } from 'components/StaffList/types';
import React from 'react';
import { connect } from 'react-redux';
import { Link, Route, Switch } from 'react-router-dom';
import AdminNewStaff from '../../Admin/NewStaff';

interface Props {
  staff: StaffMember[];
}

export class Staff extends React.Component<Props, {}> {
  render() {
    return (
      <Main>
        <PageHeader label="Staff">
          <Link to="/admin/staff/new">
            <Button>Invite Staff</Button>
          </Link>
        </PageHeader>
        <Label>Name & Email</Label>
        <StaffList staff={this.props.staff} />
        <Switch>
          <Route path="/admin/staff/new" component={AdminNewStaff} />
        </Switch>
      </Main>
    );
  }
}

const mapStateToProps = ({ staff }) => ({
  staff,
});

export default connect(mapStateToProps)(Staff);
