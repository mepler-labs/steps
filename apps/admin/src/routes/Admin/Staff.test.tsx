import { shallow } from 'enzyme';
import 'jest';
import * as React from 'react';
import { Staff } from './Staff';

describe('Staff', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Staff />);

    expect(wrapper).toMatchSnapshot();
  });
});
