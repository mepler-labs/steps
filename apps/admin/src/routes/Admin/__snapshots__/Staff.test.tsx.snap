// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staff renders correctly 1`] = `
<styled.div>
  <div
    className="header"
  >
    <h1>
      Staff
    </h1>
    <ButtonLink
      to="/admin/staff/new"
      white={false}
    >
      Invite Staff
    </ButtonLink>
  </div>
  <label>
    Name & Email
  </label>
  <Panel>
    <p>
      STUFF HERE
    </p>
    <Button
      white={false}
    >
      Edit
    </Button>
  </Panel>
  <Switch>
    <Route
      component={[Function]}
      path="/admin/staff/new"
    />
  </Switch>
</styled.div>
`;
